host/sgx/loadelf.c

```diff
@@ -319,12 +320,14 @@ static oe_result_t _stage_image_segments(
                         "Failed to get segment at index %lu",
                         i);
                 }
-
+                OE_CHECK(segment->vaddr > image->image_size);
```

Comment:
OE_CHECK expects oe_result_t

Reply:
As condition evaluates to zero, wont the above expression evaluates to OE_OK.

Reply:
@manojrupireddy , I think there might be additional warnings with this implicit cast.

Reply:
Using OE_CHECK for a conditional expression is hard to read. 
Do you expect the condition to evaluate to false?

Reply:
BTW, can you explain what's this check for?